mysql-test/suite/innodb/t/foreign_null.test

```diff
@@ -0,0 +1,110 @@
+--source include/have_innodb.inc
+let $algorithm=`select regexp_replace('$MTR_COMBINATIONS', 'innodb,\|,innodb', '')`;
+let $error_code = 0;
+
+if ($algorithm == "COPY") {
+ let $error_code = ER_FK_COLUMN_CANNOT_CHANGE;
+}
+
+if ($algorithm == "INPLACE") {
+ let $error_code = ER_FK_COLUMN_NOT_NULL;
+}
```

Comment:
Ideally it is better experience for user if he gets same error for COPY and INPLACE algorithm. Do you think we can have ER_FK_COLUMN_CANNOT_CHANGE for both ? If needed we can also raise ER_FK_COLUMN_NOT_NULL for inplace which should be stacked and shown with "show errors".